### PR TITLE
Adding a Reviung5 shield 

### PR DESCRIPTION
**Pull request by zblesk**
## Board/Shield Check-list
 - [x] This board/shield is tested working on real hardware
 - [x] Definitions follow the general style of other shields/boards upstream ([Reference](https://zmk.dev/docs/development/new-shield))
 - [x] `.zmk.yml` metadata file added
 - [x] Proper Copyright + License headers added to applicable files (Generally, we stick to "The ZMK Contributors" for copyrights to help avoid churn when files get edited)
 - [x] General consistent formatting of DeviceTree files
 - [x] Keymaps do not use deprecated key defines (Check using the [upgrader tool](https://zmk.dev/docs/codes/keymap-upgrader))
 - [x] `&pro_micro` used in favor of `&pro_micro_d/a` if applicable
 - [x] If split, no name added for the right/peripheral half
 - [x] Kconfig.defconfig file correctly wraps *all* configuration in conditional on the shield symbol
 - [x] `.conf` file has optional extra features commented out
 - [x] Keyboard/PCB is part of a shipped group buy or is generally available in stock to purchase (OSH/personal projects without general availability should create a zmk-config repo instead)

~Creating as draft until I fix the missing copyrights.~

### CODE REVIEW

File: app/boards/shields/reviung5/Kconfig.defconfig [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2020 The ZMK Contributors
```
Comment (petejohanson) [inline app/boards/shields/reviung5/Kconfig.defconfig:1]: Please fix all the copyright dates for the PR.

File: app/boards/shields/reviung5/reviung5.keymap [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+#include <behaviors.dtsi>
+#include <dt-bindings/zmk/keys.h>
+#include <dt-bindings/zmk/bt.h>
+#include <dt-bindings/zmk/outputs.h>
+
+#define BASE 0
+#define BLE 1
+
+&encoder { status = "okay"; };
```
Comment (petejohanson) [inline app/boards/shields/reviung5/reviung5.keymap:15]: No need to do this for the keymap if there's just one encoder, just enable it in the overlay where the encoder node is defined.

File: app/boards/shields/reviung5/reviung5.overlay [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    default_transform: keymap_transform_0 {
+        compatible = "zmk,matrix-transform";
+        columns = <5>;
+        rows = <1>;
+
+        map = <RC(0,0) RC(0,1) RC(0,2) RC(0,3) RC(0,4)>;
+    };
+
+    kscan0: kscan_0 {
+            compatible = "zmk,kscan-gpio-matrix";
```
Comment (petejohanson) [inline app/boards/shields/reviung5/reviung5.overlay:24]: I'm flabbergasted... It doesn't just use direct pins?

**Review by petejohanson — CHANGES_REQUESTED** [review]
Couple minor things. Thanks for the PR!

File: app/boards/shields/reviung5/reviung5.overlay (hunk 3 above)
Comment (zblesk) [reply app/boards/shields/reviung5/reviung5.overlay:24]: I'm not sure - are you asking if it really uses a matrix? Then yes. One row with five columns.

File: app/boards/shields/reviung5/reviung5.keymap (hunk 2 above)
Comment (zblesk) [reply app/boards/shields/reviung5/reviung5.keymap]: Hopefully it's okay now.

File: app/boards/shields/reviung5/reviung5.keymap [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+                // ╰─────────────┴──────────────┴──────────────────┴─────────────┴─────────────╯
+              >;
+              sensor-bindings = <&inc_dec_kp C_VOL_UP C_VOL_DN>;
+            };
+
+            ble_layer {
+              label = "BLE";
+              bindings = <
+                // ╭─────────────┬─────────────┬─────────────┬─────────────┬─────────────╮
+                     &to BASE     &out OUT_TOG   &bt BT_PRV    &bt BT_NXT    &bt BT_CLR
```
Comment (petejohanson) [inline app/boards/shields/reviung5/reviung5.keymap:33]: Should just be transparent over the mo:


```suggestion
                     &trans     &out OUT_TOG   &bt BT_PRV    &bt BT_NXT    &bt BT_CLR
```

**Review by petejohanson — CHANGES_REQUESTED** [review]
One last thing I spotted.

**Review by petejohanson — APPROVED** [review]
Thanks!

**Pull request merged**